alpha/apps/kaltura/lib/myEntryUtils.class.php

```diff
@@ -688,8 +688,74 @@ public static function createThumbnailFromEntry ( entry $entry , entry $source_e
 		}
 		return true;
 	}
-	
-	
+
+
+	public static function createThumbPaths($entity, $thumbNameAttributes, $thumbFilename, $format, $version, $thumbDirs, $contentPath, $captureRequest = false)
+	{
+		//create final path for thumbnail created
+		$finalThumbPath = myContentStorage::getThumbEntityPath(self::THUMB_ENTITY_NAME_PREFIX . $thumbDirs[0], $entity, $entity->getId() . $thumbNameAttributes, $thumbFilename, $version);
+
+		//Add unique id to the processing file path to avoid file being overwritten when several identical (with same parameters) calls are made before the final thumbnail is created
+		$uniqueThumbName = $entity->getId() . $thumbNameAttributes . "_" . uniqid() . "_";
+
+		//create path for processing thumbnail request
+		$processingThumbPath = sys_get_temp_dir() . myContentStorage::getGeneralEntityPath(self::THUMB_ENTITY_NAME_PREFIX . $thumbDirs[0], $entity->getIntId(), $uniqueThumbName, $thumbFilename, $version);
+
+		if(!is_null($format))
+		{
+			$finalThumbPath = kFile::replaceExt($finalThumbPath, $format);
+			$processingThumbPath = kFile::replaceExt($processingThumbPath, $format);
+		}
+
+		KalturaLog::debug("Path for saving capture thumbnail is [$finalThumbPath]");
+		if(kFile::checkFileExists($finalThumbPath) && @kFile::fileSize($finalThumbPath))
+		{
+			header(self::CACHED_THUMB_EXISTS_HEADER . md5($finalThumbPath));
+			return array($finalThumbPath, null);
+		}
+
+		foreach ($thumbDirs as $thumbDir)
+		{
+			$currPath = $contentPath . myContentStorage::getGeneralEntityPath(self::THUMB_ENTITY_NAME_PREFIX . $thumbDir, $entity->getIntId(), $entity->getId() . $thumbNameAttributes, $thumbFilename, $version);
+			KalturaLog::debug("Final path not found [$finalThumbPath], checking if file exists on old mount path [$currPath]");
+			if (file_exists($currPath) && @filesize($currPath))
+			{
+				if(myCloudUtils::shouldExportThumbToCloud())
+				{
+					KalturaLog::debug("File found on old mount, syncing cached thumb from [$currPath] to [$finalThumbPath], original file mtime: "
+						. date("Y-m-d H:i:s", filemtime($currPath)) );
+					$moveFileSuccess = kFile::moveFile($currPath, $finalThumbPath);
+					if($moveFileSuccess)
+					{
+						header(self::CACHED_THUMB_EXISTS_HEADER . md5($finalThumbPath));
+						return array($finalThumbPath, null);
+					}
+					else
+					{
+						KalturaLog::debug("Failed to move thumbnail from [$currPath] to [$finalThumbPath], will return oldPath");
+					}
+				}
+
+				header(self::CACHED_THUMB_EXISTS_HEADER . md5($currPath));
+				return array($currPath, null);
+			}
+		}
+
+		if ($captureRequest)
+		{
+			KalturaLog::debug("Capture thumbnail request, updating the path for saving thumbnail");
+			$captureFlavorAsset = myPackagerUtils::getFlavorSupportedByPackagerForThumbCapture($entity->getId());
```

Comment:
if you are only looking on packager supported flavor and in case you didnt find it you exit you will lose existing supported local options for example:
alpha/apps/kaltura/lib/myEntryUtils.class.php:964

Reply:
changed to prev 
$captureFlavorAsset = self::getFlavorAssetForLocalCapture($entry);
but there was a prev comment about it that is not clear 
https://github.com/kaltura/server/pull/11388#discussion_r802638693